src/storefront.ts

```diff
@@ -175,8 +175,12 @@ export class Storefront {
         this.config.pages.forEach(page => {
             const targetPage = page.url.toString();
             logger.info(`Adding page ${page.name} route: ${targetPage}`);
-            this.server.addRoute(page.url, HTTP_METHODS.GET, (req, res) => {
-                this.pages[targetPage].handle(req, res);
+            this.server.addRoute(page.url, HTTP_METHODS.GET, (req, res, next) => {
+                if (page.condition ? page.condition(req) : true) {
```

Comment:
i think it's should be `typeof page. condition === 'function'`

Reply:
Done